website/client/components/challenges/challengeModal.vue

```diff
@@ -313,6 +296,24 @@ export default {
     },
   },
   methods: {
+    async shown () {
+      this.groups = await this.$store.dispatch('guilds:getMyGuilds');
+      const party = await this.$store.dispatch('party:getParty');
+      if (party.data._id) {
```

Comment:
is party.data necessary here? Ideally `party:getParty` should return the party data directly

Reply:
Hrm, I did get the same feeling writing that code. I set it up that way because of ```asyncResourceFactory```. If ```party:getParty``` returns the party data directly, my understanding is that it also has to stash the party data directly. Looking at this code in ```store/index.js```:
```
party: {
        data: asyncResourceFactory(),
        quest: {},
        members: asyncResourceFactory(),
      },
```
How would ```party:getParty``` store the data directly without interfering with ```quest``` and ```members```? We can't just overwrite it with ```party:asyncResourceFactory(),```.

If we don't change any of this I could just do a destructuring assignment, a la:
```
const { partyId, partyName } = await this.$store.dispatch('party:getParty').data;
```
Or something along those lines, if you think that is cleaner.

Reply:
ok, I think this is super confusing because we're using the asyncResourceFactory on a sub property.

The way it works, if you had it defined like this
```
party: asyncResourceFactory(),
```

the generated structure would look like this

```

Reply:
ok, I think this is super confusing because we're using the asyncResourceFactory on a sub property.

The way it works, if you had it defined like this
```
party: asyncResourceFactory(),
```

the generated structure would look like this

```
party: {data: ..., loadingStatus: ...},
```

so the party data would be available under `store.party.data`, but since we're storing the async resource under `party.data` it'll look like this

```
party: {
    quests: ...,
    members: ...,
    data: { data: ...., loadingStatus: ... }
}
```

so the party data will be under store.party.data.data (which is indeed confusing)

but party:getParty should return directly the party data, and I think it does since the getParty action is written exactly like the fetch user action. But as you said it's not the best practice to return data from an action so instead of doing this

```
const party = await this.$store.dispatch('party:getParty');
```

You can simply do 

```
await this.$store.dispatch('party:getParty');
const party = this.$store.state.party.data.data
```

and it'll be fine :)

Reply:
I suppose I still don't understand what you mean by ```party:getParty``` returning the data directly. The last change you suggested seems to be the best option and I just pushed that. The alternative I thought of would be to use ```mapState``` for ```party.data```, await the ```party:getParty``` call and then look inside the component state. But I opted not to go that route since the party data isn't used anywhere else in the component.